### PR TITLE
[algorithmPaintTlbx] switch between algo AND cursor on Paint

### PR DESCRIPTION
**Pull request by mathildemerle**
:point_up_2: 

> 1°Switch between methods in medSegmentationSelectorToolBox is debugged. When you start to segment with Paint for instance, and you switch on polygonROI, the Paint method is deactivate. Idem polygonROI->Paint.
> 
> 2° I added a cursor during the paint, adapted to the size of the brush. 

:rotating_light:

### CODE REVIEW

File: src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp [hunk 1]
```diff
@@ -556,6 +559,47 @@ void AlgorithmPaintToolbox::activateStroke()
...
+    {
+        return;
+    }
+
+    // Get size of the brush
+    medAbstractImageView * imageView = dynamic_cast<medAbstractImageView *>(currentView);
+    int si = (int)((float)(m_brushSizeSpinBox->value()*2.0f)/imageView->sliceThickness());
+
+    // Create shape of the new cursor
+    QPixmap *pix = new QPixmap(si,si);
```
Comment (mikejbuckingham) [inline src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp]: Leak? I don't see why this needs to be heap allocated - we pass by copy when creating the QCursor on 595, and the only other use of this is as an argument to a stack-variable (painter).

File: src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp [hunk 2]
```diff
@@ -543,10 +544,12 @@ void AlgorithmPaintToolbox::updateMagicWandComputation()
...
 void AlgorithmPaintToolbox::activateStroke()
 {
-    if ( this->m_strokeButton->isChecked() ) {
+    if ( this->m_strokeButton->isChecked() )
+    {
+        deactivateCustomedCursor(); // deactivate painting cursor
         this->m_viewFilter->removeFromAllViews();
         m_paintState = (PaintState::None);
-        updateButtons();
+        updateButtons();
```
Comment (mikejbuckingham) [inline src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp]: Please check for things like this in the diffs before you make the PR.

Comment (mathildemerle) [inline src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp]: Héhé indeed

**Comment by mathildemerle** [comment]
Took into account advise of Mike :)

**Comment by mathildemerle** [comment]
Add : debug switch paint/magicWand to PolygonROI

**Comment by mathildemerle** [comment]
Going to merge it soon also. Waiting for your comments if needed :)

**Comment by LoicCadour** [comment]
I like custom cursor :)

**Comment by LoicCadour** [comment]
I like custom cursor
BUT it's not disabled when changing the workspace. We need to allow this cursor only in paint Seg (and only on the view if possible).

**Pull request merged**